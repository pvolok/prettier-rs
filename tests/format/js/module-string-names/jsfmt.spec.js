run_spec(import.meta, ["babel"], {
  errors: { meriyah: true },
});
