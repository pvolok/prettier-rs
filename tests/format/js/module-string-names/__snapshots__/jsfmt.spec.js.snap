// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`module-string-names-export.js [meriyah] format 1`] = `
"Only a identifier can be used to indicate alias (1:26)
> 1 | export { smile as "smile1" } from "./emojis.js";
    |                          ^
  2 | export { "smile" as smile2 } from "./emojis.js";
  3 | export { "smile" as "smile3" } from "./emojis.js";
  4 | export { foo1, bar as "foo2" } from "./emojis.js";"
`;

exports[`module-string-names-export.js format 1`] = `
====================================options=====================================
parsers: ["babel"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
export { smile as "smile1" } from "./emojis.js";
export { "smile" as smile2 } from "./emojis.js";
export { "smile" as "smile3" } from "./emojis.js";
export { foo1, bar as "foo2" } from "./emojis.js";
export { "學而時習之，不亦說乎？", "吾道一以貫之。" as "忠恕。" } from "Confucius";
export { "smile4" } from "./emojis.js";

=====================================output=====================================
export { smile as "smile1" } from "./emojis.js";
export { "smile" as smile2 } from "./emojis.js";
export { "smile" as "smile3" } from "./emojis.js";
export { foo1, bar as "foo2" } from "./emojis.js";
export {
  "學而時習之，不亦說乎？",
  "吾道一以貫之。" as "忠恕。",
} from "Confucius";
export { "smile4" } from "./emojis.js";

================================================================================
`;

exports[`module-string-names-import.js [meriyah] format 1`] = `
"Expected '}' (1:18)
> 1 | import { "default" as quotation1 } from "Confucius";
    |                  ^
  2 | import { "foo" as bar, "default" as qux } from "module-a";
  3 | import { "學而時習之，不亦說乎？" as quotation2 } from "Confucius";
  4 |"
`;

exports[`module-string-names-import.js format 1`] = `
====================================options=====================================
parsers: ["babel"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
import { "default" as quotation1 } from "Confucius";
import { "foo" as bar, "default" as qux } from "module-a";
import { "學而時習之，不亦說乎？" as quotation2 } from "Confucius";

=====================================output=====================================
import { "default" as quotation1 } from "Confucius";
import { "foo" as bar, "default" as qux } from "module-a";
import { "學而時習之，不亦說乎？" as quotation2 } from "Confucius";

================================================================================
`;
